test/facets/Token/behaviors/execute.t.sol

```diff
@@ -0,0 +1,144 @@
+    /// @dev Its major aim is to prevent arbitrary calls to the contract to be possible
+    /// @dev It checks wether the transaction has been approved by the AxelarGateway
+    function test_executeCannotBeCalledIfGatewayHasNotApprovedTheCall() public {
+        _enableChain(OWNER);
+
+        bytes memory payload = abi.encode(AMOUNT_TO_MINT, ALICE);
+
+        vm.expectRevert(NotApprovedByGateway.selector);
+
+        vm.startPrank(ALICE);
+        ITokenBridge(tokenAddress).execute(
+            SELECTOR_APPROVE_CONTRACT_CALL,
+            supportedChain,
+            destinationAddress,
+            payload
+        );
+    }
+
+    function test_receiveContractCallAndExecute() public {
+        _enableChain(OWNER);
+        _approveContractCall();
+
+        vm.startPrank(AXELAR_RELAYER);
+        bytes memory payload = abi.encode(AMOUNT_TO_MINT, ALICE);
+        ITokenBridge(tokenAddress).execute(
+            SELECTOR_APPROVE_CONTRACT_CALL,
+            supportedChain,
+            destinationAddress,
+            payload
+        );
+        vm.stopPrank();
+
+        uint256 amountForFees = (AMOUNT_TO_MINT *
+            token.distributionFractionBP()) / token.BASIS();
+        vm.assertEq(token.balanceOf(ALICE), AMOUNT_TO_MINT - amountForFees);
+    }
+
+    function test_callMadeFromUnsupportedAddressReverts() public {
+        _enableChain(OWNER);
+        _approveContractCallWithUnsupportedAddress();
+
+        vm.startPrank(AXELAR_RELAYER);
+        bytes memory payload = abi.encode(AMOUNT_TO_MINT, ALICE);
+        vm.expectRevert(TokenBridge__NotCorrectSourceAddress.selector);
+
+        ITokenBridge(tokenAddress).execute(
+            SELECTOR_APPROVE_CONTRACT_CALL,
+            supportedChain,
+            exploiterAddress,
+            payload
+        );
+        vm.stopPrank();
+    }
+
+    /// @notice It is a utility function to enable supported chains
+    function _enableChain(address _user) internal {
+        vm.startPrank(_user);
+        ITokenBridge(tokenAddress).enableSupportedChains(
+            supportedChain,
+            destinationAddress
+        );
+    }
+
+    /// @notice It is a utility function to approve contract calls an bypass the original signature functionality
+    function _approveContractCall() internal {
+        vm.startPrank(AXELAR_RELAYER);
+
+        bytes memory payload = abi.encode(AMOUNT_TO_MINT, ALICE);
+        bytes32 payloadHash = keccak256(payload);
+        bytes memory params = abi.encode(
+            supportedChain,
+            destinationAddress,
+            tokenAddress,
+            payloadHash,
+            keccak256("sourceTxHash"),
+            1
+        );
+        axelarGateway.approveContractCall(
+            params,
+            SELECTOR_APPROVE_CONTRACT_CALL
+        );
+    }
+    /// @notice It is a utility function to approve contract calls made from unsupported addresses
+    function _approveContractCallWithUnsupportedAddress() internal {
```

Comment:
This sets up to check that even if the call is approved by relayer, when the source address is not approved, call reverts?

Reply:
exactly